openedx/adg/common/djangoapps/user_authn_override/views/registration_form.py

```diff
@@ -0,0 +1,161 @@
+"""
+Overridden RegistrationFormFactory from openedx.core.djangoapps.user_authn.views.registration_form.py
+"""
+
+
+from django.utils.translation import ugettext as _
+
+from edxmako.shortcuts import marketing_link
+from openedx.adg.constants import ADG_BUSINESS_LINES, CITIES_OF_KSA
+from openedx.core.djangoapps.user_api import accounts
+from openedx.core.djangoapps.user_authn.views.registration_form import (
+    RegistrationFormFactory as CoreRegistrationFormFactory
+)
+from openedx.core.djangolib.markup import HTML, Text
+
+
+class RegistrationFormFactory(CoreRegistrationFormFactory):
+    """
+    Overridden openedx RegistrationFormFactory to avoid core changes.
+    """
+
+    EXTRA_FIELDS = CoreRegistrationFormFactory.EXTRA_FIELDS + ["is_adg_employee"]
+
+    def _add_username_field(self, form_desc, required=True):
```

Comment:
I guess we override this, to just change the label? I believe there are some other ways to change the label as well. We can access the fields in __init__ and set the label, or we can use the labels attrs of django forms.
This is applicable for the fields where overrided the whole method to just change the label.

Reply:
Or if we want to change some other things as well, please think about some way to achieve it without overriding the complete method, if no way then we should override the whole method.

Reply:
I did this for the fields where this should have been this way. I have avoided this for 2 fields city and company as there is a complete logic for select in the add_field method and I need that logic to be applied on city and company for the selects.